test/core/tsi/alts/crypt/aes_gcm_test.cc

```diff
@@ -2093,13 +2094,85 @@ static void gsec_test_do_vector_tests_ieee() {
   gsec_aead_free_test_vector(test_vector_20);
 }
 
+static void hkdf_check_out_key(GsecHashFunction hash_function,
+                               std::vector<uint8_t>& info,
+                               std::vector<uint8_t>& prk,
+                               std::vector<uint8_t>& correct_out_key) {
+  std::vector<uint8_t> out_key;
+  out_key.resize(correct_out_key.size(), 0);
+  grpc_status_code derive_status =
+      hkdf_derive_secret(out_key.data(), out_key.size(), hash_function,
+                         prk.data(), prk.size(), info.data(), info.size());
+  GPR_ASSERT(derive_status == GRPC_STATUS_OK);
+  for (size_t i = 0; i < out_key.size(); i++) {
+    GPR_ASSERT(out_key[i] == correct_out_key[i]);
```

Comment:
Can you use out_key == correct_out_key now?

Reply:
Done.